elementary/monitor/dbt_project/macros/base_queries/current_tests_run_results_query.sql

```diff
@@ -26,248 +12,57 @@
-        test_results.status,
-        test_results.test_short_name,
-        test_results.test_alias,
-        test_results.result_rows,
-        tests.meta,
-        first_occurred.first_time_occurred as test_created_at
-    from elementary_test_results_with_final_unique_id test_results
-    left join first_time_test_occurred first_occurred on test_results.elementary_unique_id = first_occurred.elementary_unique_id
-    left join dbt_tests_with_final_unique_id tests on test_results.elementary_unique_id = tests.elementary_unique_id
-{% endmacro %}
-
-
-{# The use of this macro is to support old versions of elementary dbt package #}
-{% macro current_tests_run_results_backcomp(days_back = none, invocation_id = none) %}
-    with elementary_test_results as (
-        select * from {{ ref('elementary_test_results') }}
-        {% if invocation_id %}
-            where invocation_id = {{ "'{}'".format(invocation_id) }}
-        {% elif days_back %}
-            where {{ elementary.datediff(elementary.cast_as_timestamp('detected_at'), elementary.current_timestamp(), 'day') }} < {{ days_back }}
-        {% endif %}
-    ),
-    
-    dbt_tests as (
-        select * from {{ ref('dbt_tests') }}
-    ),
-    
     first_time_test_occurred as (
         select 
             min(detected_at) as first_time_occurred,
             test_unique_id
         from elementary_test_results
         group by test_unique_id
     )
-    
+
     select
-        test_results.id,
-        test_results.data_issue_id,
-        test_results.test_execution_id,
-        test_results.test_unique_id,
-        test_results.model_unique_id,
-        test_results.detected_at,
-        test_results.database_name,
-        test_results.schema_name,
-        test_results.table_name,
-        test_results.column_name,
-        test_results.test_type,
-        test_results.test_sub_type,
-        test_results.test_results_description,
-        test_results.owners,
-        test_results.tags,
-        test_results.test_results_query,
-        test_results.other,
-        test_results.test_name,
-        test_results.test_params,
-        test_results.severity,
-        test_results.status,
-        tests.meta,
+        elementary_test_results.id,
+        {#
+            Due to inconsistency test sub type and column name in some of elementary's tests
+            we need to calculate different test sub type unique id which is used to identify between different tests.
+        #}
+        case
+            when elementary_test_results.test_type = 'schema_change' then elementary_test_results.test_unique_id
+            {# In old versions of elementary, elementary_test_results doesn't contain test_short_name, so we use dbt_test short_name. #}
+            when dbt_tests.short_name = 'dimension_anomalies' then elementary_test_results.test_unique_id
+            else coalesce(elementary_test_results.test_unique_id, 'None') || '.' || coalesce(elementary_test_results.test_sub_type, 'None')
```

Comment:
if it's a dbt test we should use the test_unique_id, no?

Reply:
The subtype of dbt test is always the same + each test is mapped to exactly one elementary_test_results, so it is fine to use the `test_unique_id` + `test_sub_type` for tests that aren't ours.
One key generation logic to rule them all!
(It is simpler and better IMO)